pkg/clients/resourcerecordset/resourcerecordset.go

```diff
@@ -0,0 +1,173 @@
+					},
+				},
+			},
+		},
+	}
+}
+
+// IsUpToDate checks if object is up to date
+func IsUpToDate(p v1alpha3.ResourceRecordSetParameters, rrset route53.ResourceRecordSet) (bool, error) {
+	// check for the root "." found in DNS entries and add if not found
+	if !strings.HasSuffix(*p.Name, ".") {
+		p.Name = aws.String(fmt.Sprintf("%s.", *p.Name))
+	}
+	patch, err := CreatePatch(&rrset, &p)
+	if err != nil {
+		return false, err
+	}
+	return cmp.Equal(&v1alpha3.ResourceRecordSetParameters{}, patch), nil
+}
+
+// LateInitialize fills the empty fields in *v1alpha3.ResourceRecordSetParameters with
+// the values seen in route53.ResourceRecordSet.
+func LateInitialize(in *v1alpha3.ResourceRecordSetParameters, rrSet *route53.ResourceRecordSet) {
+	if rrSet == nil {
+		return
+	}
+
+	in.Name = awsclients.LateInitializeStringPtr(in.Name, rrSet.Name)
+
+	rrType := string(rrSet.Type)
+	in.Type = awsclients.LateInitializeStringPtr(in.Type, &rrType)
+
+	in.TTL = awsclients.LateInitializeInt64Ptr(in.TTL, rrSet.TTL)
+	if len(in.Records) == 0 && len(rrSet.ResourceRecords) != 0 {
+		in.Records = make([]string, len(rrSet.ResourceRecords))
+		for i, val := range rrSet.ResourceRecords {
+			in.Records[i] = aws.StringValue(val.Value)
+		}
+	}
+}
+
+// CreatePatch creates a *v1beta1.ResourceRecordSetParameters that has only the changed
+// values between the target *v1beta1.ResourceRecordSetParameters and the current
+// *route53.ResourceRecordSet
+func CreatePatch(in *route53.ResourceRecordSet, target *v1alpha3.ResourceRecordSetParameters) (*v1alpha3.ResourceRecordSetParameters, error) {
+	currentParams := &v1alpha3.ResourceRecordSetParameters{}
+	LateInitialize(currentParams, in)
+
+	jsonPatch, err := awsclients.CreateJSONPatch(currentParams, target)
+
+	if err != nil {
+		return nil, err
+	}
+	patch := &v1alpha3.ResourceRecordSetParameters{}
+	if err := json.Unmarshal(jsonPatch, patch); err != nil {
+		return nil, err
+	}
+	patch.ZoneID = patch.Name
+	return patch, nil
+}
+
+// GetResourceRecordSet returns recordSet if present or err
+func GetResourceRecordSet(ctx context.Context, c Client, p v1alpha3.ResourceRecordSetParameters) (route53.ResourceRecordSet, error) {
+	// check for the root "." found in DNS entries and add if not found
+	if !strings.HasSuffix(*p.Name, ".") {
+		p.Name = aws.String(fmt.Sprintf("%s.", *p.Name))
+	}
+	req := c.ListResourceRecordSetsRequest(&route53.ListResourceRecordSetsInput{
+		HostedZoneId: p.ZoneID,
+	})
+	res, err := req.Send(ctx)
+	if err != nil {
+		return route53.ResourceRecordSet{}, err
+	}
+
+	for _, rrSet := range res.ResourceRecordSets {
+		if aws.StringValue(rrSet.Name) == aws.StringValue(p.Name) && string(rrSet.Type) == aws.StringValue(p.Type) {
+			return rrSet, nil
+		}
+	}
+
+	return route53.ResourceRecordSet{}, &NotFound{}
+}
+
+// StringValueFromRRType returns the string value of aws ResourceRecordType
+// passed in or "" if nil
+func StringValueFromRRType(rrType route53.RRType) string {
```

Comment:
This function doesn't seem to be used anywhere.

Reply:
That is correct and hence the function has been removed.